cpp/perspective/src/include/perspective/gnode_state.h

```diff
@@ -177,11 +203,18 @@ class PERSPECTIVE_EXPORT t_gstate {
     t_tscalar get_value(const t_tscalar& pkey, const std::string& colname) const;
 
     /**
-     * @brief Return the size of the underlying `t_data_table`.
+     * @brief Return the number of rows on the master `t_data_table`.
+     * 
+     * @return t_uindex 
+     */
+    t_uindex num_rows() const;
+
+    /**
+     * @brief Return the number of columns on the master `t_data_table`.
      * 
      * @return t_uindex 
      */
-    t_uindex size() const;
+    t_uindex num_columns() const;
```

Comment:
We may as well go all the way and apply this change to `Table.size()` API in JS and Python!